Documentation/Proposals/NNNN-exit-tests.md

```diff
@@ -0,0 +1,787 @@
+  ...
+}
+```
+
+Today, a test author can write unit tests for this function, but there is no way
+to make sure that the function rejects a taco whose `isDelicious` property is
+`false` because a test that passes such a taco as input will crash (correctly!)
+when it calls `precondition()`.
+
+An exit test allows testing this sort of functionality. The mechanism by which
+an exit test is implemented varies between testing libraries and languages, but
+a common implementation involves spawning a new process, performing the work
+there, and checking that the spawned process ultimately terminates with a
+particular (possibly platform-specific) exit status.
+
+Adding exit tests to Swift Testing would allow an entirely new class of tests
+and would improve code coverage for existing test targets that adopt them.
+
+## Proposed solution
+
+This proposal introduces new overloads of the `#expect()` and `#require()`
+macros that take, as an argument, a closure to be executed in a child process.
+When called, these macros spawn a new process using the relevant
+platform-specific interface (`posix_spawn()`, `CreateProcessW()`, etc.), call
+the closure from within that process, and suspend the caller until that process
+terminates. The exit status of the process is then compared against a known
+value passed to the macro, allowing the test to pass or fail as appropriate.
+
+The function from earlier can then be tested using either of the new
+overloads:
+
+```swift
+await #expect(exitsWith: .failure) {
+  var taco = Taco()
+  taco.isDelicious = false
+  eat(taco) // should trigger a precondition failure and process termination
+}
+```
+
+## Detailed design
+
+### New expectations
+
+We will introduce the following new overloads of `#expect()` and `#require()` to
+the testing library:
+
+```swift
+/// Check that an expression causes the process to terminate in a given fashion.
+///
+/// - Parameters:
+///   - expectedExitCondition: The expected exit condition.
+///   - observedValues: An array of key paths representing results from within
+///     the exit test that should be observed and returned by this macro. The
+///     ``ExitTest/Result/statusAtExit`` property is always returned.
+///   - comment: A comment describing the expectation.
+///   - sourceLocation: The source location to which recorded expectations and
+///     issues should be attributed.
+///   - expression: The expression to be evaluated.
+///
+/// - Returns: If the exit test passed, an instance of ``ExitTest/Result``
+///   describing the state of the exit test when it exited. If the exit test
+///   fails, the result is `nil`.
+///
+/// Use this overload of `#expect()` when an expression will cause the current
+/// process to terminate and the nature of that termination will determine if
+/// the test passes or fails. For example, to test that calling `fatalError()`
+/// causes a process to terminate:
+///
+/// await #expect(exitsWith: .failure) {
+///   fatalError()
+/// }
```

Comment:
Either indent or use backticks around this code listing, otherwise it will render as a paragraph.

Reply:
This is a code block inside a code block. It doesn't need to be rendered.